src/frontend/src/flows/verifiableCredentials/allowCredentials.ts

```diff
@@ -8,12 +8,33 @@ import { html, TemplateResult } from "lit-html";
 import { asyncReplace } from "lit-html/directives/async-replace.js";
 import { unsafeHTML } from "lit-html/directives/unsafe-html.js";
 
+import { getDapps, KnownDapp } from "../dappsExplorer/dapps";
+import { dappTemplate } from "../dappsExplorer/index";
+
 import DOMPurify from "dompurify";
 
 import copyJson from "./allowCredentials.json";
 
 /* A screen prompting the user to allow (or cancel) issuing verified
  * credentials */
+const getOrigin = (
+  origin: string,
+  dapplist: KnownDapp[],
+  oneLiner: string
+): KnownDapp => {
+  let foundDapp = dapplist.find((dapp) => dapp.hasOrigin(origin));
+
+  if (!foundDapp) {
+    foundDapp = new KnownDapp({
+      name: "Unknown Dapp",
+      website: origin,
+      logo: "unwnowndapp.png",
+      oneLiner: oneLiner,
+    });
+  }
+  foundDapp.oneLiner = oneLiner;
```

Comment:
Nit: I think it is slightly hacky to override the `oneLiner` with the `copy` string on the `KnownDapp`.
What about overriding the value in the call to `dappsTemplate`?

Reply:
@frederikrothenberger you are right, I think its better now. let me know if this works for you